libraries/src/test/java/com/baeldung/smack/SmackIntegrationTest.java

```diff
@@ -0,0 +1,92 @@
+package com.baeldung.smack;
+
+import org.jivesoftware.smack.AbstractXMPPConnection;
+import org.jivesoftware.smack.SmackException;
+import org.jivesoftware.smack.XMPPException;
+import org.jivesoftware.smack.chat2.ChatManager;
+import org.jivesoftware.smack.filter.StanzaTypeFilter;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smack.tcp.XMPPTCPConnection;
+import org.jivesoftware.smack.tcp.XMPPTCPConnectionConfiguration;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.jxmpp.stringprep.XmppStringprepException;
+
+import java.io.IOException;
+
+public class SmackIntegrationTest {
+
+    static AbstractXMPPConnection connection;
+
+    @BeforeClass
+    public static void setup() throws IOException, InterruptedException, XMPPException, SmackException {
+
+        XMPPTCPConnectionConfiguration config = XMPPTCPConnectionConfiguration.builder()
+                .setUsernameAndPassword("baeldung","baeldung")
+                .setXmppDomain("jabb3r.org")
+                .setHost("jabb3r.org")
+                .build();
+
+        XMPPTCPConnectionConfiguration config2 = XMPPTCPConnectionConfiguration.builder()
+                .setUsernameAndPassword("baeldung2","baeldung2")
+                .setXmppDomain("jabb3r.org")
+                .setHost("jabb3r.org")
+                .build();
+
+        connection = new XMPPTCPConnection(config);
+        connection.connect();
+        connection.login();
+
+    }
+
+    @Test
+    public void whenSendMessageWithChatThenReceiveMessage() throws XmppStringprepException, InterruptedException {
```

Comment:
Thank you for using the when/then convention. This is easy to miss, but would you please do it with an underscore, like whenXXX_thenYYY, please?